python_translators/translators/composite_parallel_translator.py

```diff
@@ -0,0 +1,50 @@
+import asyncio
+import threading
+
+from python_translators.translators.translator import Translator
+from python_translators.translation_query import TranslationQuery
+from python_translators.translation_response import TranslationResponse
+from python_translators.translation_costs import TranslationCosts
+from python_translators.translation_response import merge_translations
+from python_translators.translators.composite_translator import CompositeTranslator
+
+
+def translate_worker(translator: Translator, query: TranslationQuery, responses: [TranslationResponse], idx: int):
+    responses[idx] = translator.translate(query)
+
+
+class CompositeParallelTranslator(CompositeTranslator):
+    def __init__(self, source_language: str, target_language: str, translators: [Translator] = None):
+        super(CompositeParallelTranslator, self).__init__(source_language, target_language, translators)
+
+    def _translate(self, query: TranslationQuery) -> TranslationResponse:
+        responses = [None] * len(self.translators)
+        threads = []
+
+        # Start a thread for each translator
+        for idx, translator in enumerate(self.translators):
+            t = threading.Thread(target=translate_worker, args=(translator, query, responses, idx))
+            t.start()
+            threads.append(t)
+
+        # Wait for all threads to complete
```

Comment:
nevermind, i forgot that the translation query knows about the max translations and that the dispatching happens inside translate. it would not make a difference and it still makes a bit more sense the way ti's now. 

abstracting a strategy class: maybe a translator is a strategy? 

e.g. this would be a TimeQualityCompromise translator that would have the following strategy: 
* return if: 
- the timeout has been reached (this would be something like 2s, or so...  a time after which it's not worth waiting)
- a ideal_time_timeout has been reached (this would be something like 0.75s.) and at least half of the sub-translators have returned.

et.c